roseus_smach/src/state-machine.l

```diff
@@ -134,10 +136,11 @@
          (when (not (eq astate last-state))
            (error "active state has not changed ~A -> ~A~%" last-state astate)
            (return-from :execute-impl (send astate :name)))
-         (setq active-state (send-all trans :to))
+         (setq tmp-active-state (append tmp-active-state (send-all trans :to)))
          (if (send astate :submachine)
              (send (send astate :submachine) :reset-state))
          (push exec-result ret)))
+     (setq active-state tmp-active-state)
```

Comment:
I think, we should check the uniqueness of next states in case of joining parallel states.
I recommend to use like: `(setq active-state (unique next-active-state))`.